envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/utils/GzipUtils.kt

```diff
@@ -0,0 +1,42 @@
+package pl.allegro.tech.servicemesh.envoycontrol.utils
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import pl.allegro.tech.servicemesh.envoycontrol.logger
+import java.io.ByteArrayOutputStream
+import java.lang.reflect.Type
+import java.util.zip.GZIPInputStream
+import java.util.zip.GZIPOutputStream
+import kotlin.text.Charsets.UTF_8
+
+class GzipUtils(val objectMapper: ObjectMapper) {
+
+    private val logger by logger()
+
+    fun gzip(content: Any): ByteArray {
```

Comment:
Maybe instead of implementing gzipping response by your own, we can configure spring server to do it automatically - 
```
server.compression.enabled=true
server.compression.mime-types=text/html,text/xml,text/plain,text/css,text/javascript,application/javascript,application/json
server.compression.min-response-size=1024
```
Thanks that we can ommit creating different Cotrollers  depending on `gzip` property

Reply:
done